### PR TITLE
Update to SpiderMonkey 66

### PR DESCRIPTION
**Pull request by jdm**
This change is [Reviewable](https://reviewable.io/reviews/servo/mozjs/172)

### CODE REVIEW

**Comment by asajeffrey** [comment]
This looks pretty sensible so far! OMG are we now the point where an SM upgrade takes a few days rather months?

**Comment by asajeffrey** [comment]
Hmm, it is very easy to press the "close and comment" button by mistake. [Looks sheepish.]

File: Cargo.toml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 name = "mozjs_sys"
 description = "System crate for the Mozilla SpiderMonkey JavaScript engine."
 repository = "https://github.com/servo/mozjs/"
-version = "0.61.7"
+version = "0.62.0"
```
Comment (atouchet) [inline Cargo.toml]: I think previous updates used the same version number as SM (i.e. 0.66.0). Are you not going to keep using that versioning system?

Comment (jdm) [reply Cargo.toml]: We gave up on that previously. The last update was 50-something iirc.

Comment (atouchet) [reply Cargo.toml]: The SM60 update went from 0.51.4 to 0.60.0.

Comment (asajeffrey) [reply Cargo.toml]: I was hoping to set the version to be the max of the previous version + 1 and the SM version, so it keeps in rough step with the SM version.

**Comment by jdm** [comment]
The appveyor branch build seemed to stop in the middle of running bindgen, after running for 43 minutes total? Very confusing.

**Comment by jdm** [comment]
Nevermind, it looks like it timed out after 1 hour, not 43 minutes.

**Comment by jdm** [comment]
Second windows job finished with 8 seconds to spare 😓

**Comment by jdm** [comment]
Latest commit made mac complain about -lmozglue as well as linux.

**Comment by jdm** [comment]
Windows build is missing NSPR symbols.
Linux build is missing mozglue.
Mac build is missing mozglue.

**Comment by jdm** [comment]
Note to self: mozjs/js/src/frontend/binsource/Cargo.toml is important

**Comment by jdm** [comment]
Linux and Mac builds are working.
Windows x86 build is complaining that the wrong rust libraries are being linked.

**Comment by jdm** [comment]
* Linux builds completed successfully
* mac builds completed successfully
* Windows x86 build completed successfully
* windows x64 build in progress

**Comment by jdm** [comment]
This is ready for review.

**Comment by asajeffrey** [comment]
Sorry about that.

**Comment by asajeffrey** [comment]
This all LGTM. Took a couple of months, which is better than last time but still not the few days I was hoping for. Rebase, make sure we don't lose the changes from #180 and r=me.

**Comment by jdm** [comment]
To be fair, I abandoned it for a while. I think it was less than three weeks total.

**Pull request merged**